vertx-grpc-server/src/main/java/io/vertx/grpc/server/impl/GrpcServerImpl.java

```diff
@@ -56,6 +69,26 @@ public void handle(HttpServerRequest httpRequest) {
     }
   }
 
+  private boolean refuseRequest(HttpServerRequest request) {
+    if (request.version() != HttpVersion.HTTP_2) {
```

Comment:
Is there a vert.x specific reason for not allowing gRPC-Web over HTTP/2 ?
I couldn't find a reference to this restriction in the [protocol specification](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md#protocol-differences-vs-grpc-over-http2), and Envoy's filter doesn't seem to refuse HTTP/2 request for gRPC-Web.

Reply:
what would be the use case of using gRPC web over HTTP/2 ?

Reply:
> Is there a vert.x specific reason for not allowing gRPC-Web over HTTP/2 ?

@kureuil  Not really. I did not give enough though to this, I guess.

> what would be the use case of using gRPC web over HTTP/2 ?

@vietj gRPC in the browser over HTTP/2.

The browser does not expose protocol details, so even if the client and server communicate over HTTP/2, standard gRPC cannot be used and gRPC-Web is mandatory.

Reply:
Thanks @kureuil for bringing this up

Reply:
my question was implicitely: given that gRPC over HTTP/2 is not possible in the browser, what would be the other use case of gRPC Web over HTTP/2 that is not the browser :-)

Reply:
I only have second-hand knowledge on the matter, but one other use case for gRPC-Web over HTTP/2 would be iOS applications : the standard URLSession doesn't support HTTP trailers. Applications communicating with a gRPC server usually include the SwiftNIO package, adding more than 2Mo of compiled code and a more complex network stack "just" to support trailers in the common case. The gRPC-Web protocol allows using the standard network stack, leveraging the battery- and network-savings optimizations included in the operating system.

The use case that prompted my question is gRPC Web over HTTP/2 in the browser :)